jkube-kit/resource/helm/src/main/java/org/eclipse/jkube/kit/resource/helm/HelmUploader.java

```diff
@@ -0,0 +1,155 @@
+        break;
+      case CHARTMUSEUM:
+        connection = getConnectionForUploadToChartmuseum(repository);
+        break;
+      case NEXUS:
+        connection = getConnectionForUploadToNexus(file, repository);
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported repository type for upload.");
+    }
+
+    writeFileOnConnection(file, connection);
+
+    if (connection.getResponseCode() >= 300) {
+      String response;
+      if (connection.getErrorStream() != null) {
+        response = IOUtils.toString(connection.getErrorStream(), Charset.defaultCharset());
+      } else if (connection.getInputStream() != null) {
+        response = IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
+      } else {
+        response = "No details provided";
+      }
+      throw new BadUploadException(response);
+    } else {
+      String message = Integer.toString(connection.getResponseCode());
+      if (connection.getInputStream() != null) {
+        message += " - " + IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
+      }
+      logger.info(message);
+    }
+    connection.disconnect();
+  }
+
+  protected void writeFileOnConnection(File file, HttpURLConnection connection) throws IOException {
+    try (FileInputStream fileInputStream = new FileInputStream(file)) {
+      IOUtils.copy(fileInputStream, connection.getOutputStream());
+    }
+  }
+
+  protected HttpURLConnection getConnectionForUploadToChartmuseum(HelmRepository repository) throws IOException {
+    final HttpURLConnection connection = createConnection(repository.getUrl());
+    connection.setDoOutput(true);
+    connection.setRequestMethod("POST");
+    connection.setRequestProperty("Content-Type", "application/gzip");
+
+    verifyAndSetAuthentication(repository);
+
+    return connection;
+  }
+
+  protected HttpURLConnection getConnectionForUploadToArtifactory(
+      File file,
+      HelmRepository repository
+  ) throws IOException {
+    String uploadUrl = repository.getUrl();
+    // Append slash if not already in place
+    if (!uploadUrl.endsWith("/")) {
+      uploadUrl += "/";
+    }
+    uploadUrl = uploadUrl + file.getName();
+
+    final HttpURLConnection connection = createConnection(uploadUrl);
+    connection.setDoOutput(true);
+    connection.setRequestMethod("PUT");
+    connection.setRequestProperty("Content-Type", "application/gzip");
+
+    verifyAndSetAuthentication(repository);
+
+    return connection;
+  }
+
+  protected HttpURLConnection getConnectionForUploadToNexus(File file, HelmRepository repository) throws IOException {
+    String uploadUrl = repository.getUrl();
+    // Append slash if not already in place
+    if (!uploadUrl.endsWith("/")) {
+      uploadUrl += "/";
+    }
+    uploadUrl = uploadUrl + file.getName();
+
+    final HttpURLConnection connection = createConnection(uploadUrl);
+    connection.setDoOutput(true);
+    connection.setRequestMethod("PUT");
```

Comment:
`getConnectionForUploadToNexus` and `getConnectionForUploadToArtifactory` look quite similar. Is it possible to move common code to some other method in order to avoid duplication?

Reply:
I pushed some refactoring